app/controllers/cases/data_requests_controller.rb

```diff
@@ -53,9 +50,9 @@ def update
         redirect_to case_path(@case)
       when :unprocessed
         flash[:notice] = t('.unprocessed')
-        redirect_to edit_case_data_request_path(@case, @data_request)
+        redirect_to case_path(@case)
       when :error
-        @data_request_log = service.data_request_log
+        @data_request = service.data_request
```

Comment:
I just realised you removed all those parts related to data_request_logs,   in that case I don't think I need to insert any data into this table